src/main/java/org/beanpod/switchboard/controller/EncoderController.java

```diff
@@ -110,18 +111,19 @@
     return ResponseEntity.ok("Encoder with serial number " + serialNumber + " Deleted");
   }
 
-  @PutMapping
-  public ResponseEntity<EncoderDto> updateEncoder(@RequestBody EncoderDto encoderDto) {
+  @Override
+  public ResponseEntity<EncoderModel> updateEncoder(@Valid EncoderModel encoderModel) {
     UserEntity user = userDao.findUser(request.getUserPrincipal().getName());
 
-    Optional<EncoderDto> encoder = encoderDao.findEncoder(user, encoderDto.getSerialNumber());
-    if (encoder.isEmpty()) {
-      throw new ExceptionType.DeviceNotFoundException(encoderDto.getSerialNumber());
-    }
-    return ResponseEntity.ok(encoderDao.save(user, encoderDto));
+    return encoderDao
+        .findEncoder(user, encoderModel.getSerialNumber())
+        .map((encoderDto) -> encoderDao.save(user, encoderDto))
```

Comment:
Can you please get rid of your one code smell here:
```suggestion
        .map(encoderDto -> encoderDao.save(user, encoderDto))
```

Reply:
alright! it's fixed